### PR TITLE
stratisd_cert: Add tearDown assert for specific private devices

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related https://github.com/stratis-storage/project/issues/604

### CODE REVIEW

**Review by mulkieran — CHANGES_REQUESTED** [review]
```maxDiff``` is class attribute of ```UnitTest.TestCase```. We should take the advice and set it. There are probably three ways we can set it.

I'ld recommend the simplest, it looks like this, roughly:

```
@@ -1249,6 +1253,7 @@ def main():
     parsed_args, unittest_args = argument_parser.parse_known_args()
     StratisCertify.DISKS = parsed_args.DISKS
     StratisCertify.monitor_dbus = parsed_args.monitor_dbus
+    StratisCertify.maxDiff = None
     print(f"Using block device(s) for tests: {StratisCertify.DISKS}")
     unittest.main(argv=sys.argv[:1] + unittest_args)
```

The other two options are defining an ```__init__``` method or setting it in the tearDown() method (which is kind of inelegant). So, let's try the simple approach first and see what it brings.

**Comment by bgurney-rh** [comment]
Here's the result of `test_pool_create_after_cache` after setting maxDiff to None:
```
    self.assertEqual(non_stratis_by_id_devices, [])
AssertionError: Lists differ: ['dm-name-stratis-1-private-2f5eedbf86b245[551 chars]eta'] != []

First list contains 8 additional elements.
First extra element 0:
'dm-name-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-thinpool-pool'

+ []
- ['dm-name-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-thinpool-pool',
-  'dm-uuid-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-thinpool-pool',
-  'dm-uuid-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-mdv',
-  'dm-name-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-mdv',
-  'dm-name-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-thindata',
-  'dm-uuid-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-thindata',
-  'dm-uuid-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-thinmeta',
-  'dm-name-stratis-1-private-2f5eedbf86b245b28517c8aff8775dee-flex-thinmeta']

```

**Comment by mulkieran** [comment]
> 

It looks like the one you want to check is, e.g., ```test_pool_add_cache```.  It seems like, for some reason, ```test_pool_create_after_cache```'s diff did not exceed ```unittest.TestCase```'s upper bound.

**Comment by bgurney-rh** [comment]
Result from `test_pool_add_cache`:
```
First list contains 12 additional elements.
First extra element 0:
'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-thinpool-pool'

+ []
- ['dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-thinpool-pool',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-thinpool-pool',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-mdv',
-  'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-mdv',
-  'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-physical-cache',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-physical-cache',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-physical-cachesub',
-  'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-physical-cachesub',
-  'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-thindata',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-thindata',
-  'dm-uuid-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-thinmeta',
-  'dm-name-stratis-1-private-17a61104c3ea4b46a8a58e1485afd7de-flex-thinmeta']
```

File: stratisd_cert.py [hunk 1]
```diff
@@ -235,6 +236,18 @@ def tearDown(self):
         D-Bus trace.
         :return: None
         """
+        try:
+            by_id_devices = os.listdir("/dev/disk/by-id")
+            non_stratis_by_id_devices = []
+            for dev in by_id_devices:
```
[the remark this answers is not in the document]
Comment (bgurney-rh) [reply stratisd_cert.py]: Resolved.

**Comment by bgurney-rh** [comment]
Rebased, and squashed into a single commit (before next series of fixes).

**Comment by bgurney-rh** [comment]
I chose the argument `--verify-devices`, since it seems like a concise way to describe the optional test.

Without the option, a test that passed before this PR and failed after, now passes:
```
$ sudo python3 stratisd_cert.py StratisdCertify.test_pool_create_after_cache -v --disk /dev/vdb1 --disk /dev/vdb2 --disk /dev/vdb3
Using block device(s) for tests: ['/dev/vdb1', '/dev/vdb2', '/dev/vdb3']
test_pool_create_after_cache (__main__.StratisdCertify.test_pool_create_after_cache)
Test creating existing pool after cache was added ... ok

----------------------------------------------------------------------
Ran 1 test in 5.988s

OK
```

...and with `--verify-devices`, now fails:
```
$ sudo python3 stratisd_cert.py StratisdCertify.test_pool_create_after_cache -v --verify-devices --disk /dev/vdb1 --disk /dev/vdb2 --disk /dev/vdb3
Using block device(s) for tests: ['/dev/vdb1', '/dev/vdb2', '/dev/vdb3']
test_pool_create_after_cache (__main__.StratisdCertify.test_pool_create_after_cache)
Test creating existing pool after cache was added ... FAIL

...

AssertionError: Lists differ: ['dm-uuid-stratis-1-private-6f39e1a59cf143[551 chars]eta'] != []

First list contains 8 additional elements.
First extra element 0:
'dm-uuid-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-thinpool-pool'

+ []
- ['dm-uuid-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-thinpool-pool',
-  'dm-name-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-thinpool-pool',
-  'dm-uuid-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-mdv',
-  'dm-name-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-mdv',
-  'dm-name-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-thindata',
-  'dm-uuid-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-thindata',
-  'dm-name-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-thinmeta',
-  'dm-uuid-stratis-1-private-6f39e1a59cf143d993a92881545e4da0-flex-thinmeta']

----------------------------------------------------------------------
Ran 1 test in 6.095s

FAILED (failures=1)

```

**Review by mulkieran — COMMENTED** [review]
This looks good to me. I want to wait to merge it just a bit, until we find out a little bit more on these errors.

File: stratisd_cert.py [hunk 2]
```diff
@@ -235,6 +236,18 @@ def tearDown(self):
         D-Bus trace.
         :return: None
         """
+        if StratisdCertify.verify_devices is True:
+            try:
+                non_stratis_by_id_devices = []
```
Comment (mulkieran) [inline stratisd_cert.py:241]: Please rename this to something a bit more generic, like "disallowed_syslinks" or something.

Comment (bgurney-rh) [reply stratisd_cert.py]: OK; renamed.  I'm going to let CI run, and then if it passes, squash everything into a single commit.

**Review by mulkieran — APPROVED** [review]
~~Looks good to me. Please sqash into two or one commits as you think best, and I'll merge.~~ Just saw your message. Excellent.

**Pull request merged**